### PR TITLE
Merge WasmExpressionBuilder and WasmFunctionContext into FunctionEmitter.

### PR DESCRIPTION
**Pull request by sjrd**
The combined class has the same responsibilities as the `FunctionEmitter` in the JS backend.

~~Based on #124, but otherwise ready to review.~~

### CODE REVIEW

**Comment by sjrd** [comment]
Rebased.

File: wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/ClassEmitter.scala [hunk 1]
```diff
@@ -784,144 +750,127 @@ class ClassEmitter(coreSpec: CoreSpec) {
...
+      }
+
+      def genLoadIsolatedTree(tree: IRTrees.Tree): Unit = {
+        tree match {
+          case IRTrees.StringLiteral(value) =>
+            // Common shape for all the `nameTree` expressions
+            instrs ++= ctx.getConstantStringInstr(value)
+
+          case IRTrees.VarRef(IRTrees.LocalIdent(localName))
+              if classCaptureParamsOfTypeAny.contains(localName) =>
```
Comment (tanishiking) [inline wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/ClassEmitter.scala:818]: Why we need this condition in this case?

**Review by tanishiking — APPROVED** [review]
one question, but overall LGTM :tada:

File: wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/ClassEmitter.scala (hunk 1 above)
Comment (tanishiking) [reply wasm/src/main/scala/org/scalajs/linker/backend/wasmemitter/ClassEmitter.scala:818]: > we would need adapt to fix up the type,

Ah, right, thanks!

**Pull request merged**